### DISCUSSION TITLE
What are the hardware requirements for running the checks?

### DISCUSSION THREAD

**Discussion by m1mir** [nix-community/nixvim#1810] [Q&A] [answered]
My attempt at running the tests failed because I ran out of memory, on a system that had 16 gigabytes of RAM.

Approximately what are the hardware requirements for testing the project the way it is described in the CONTRIBUTING.md?

**Comment by GaetanLepage** [comment]
Yes, unfortunately the tests are quite heavy to run.
First, ensure to only run the `tests` command and not the `checks` (which builds even more things).
Second, try adding the `--max-jobs 1` flag to limit the number of simultaneous builds (although your crash might happen earlier in the eval phase).

Finally, you could rely on the CI to run the tests by creating a PR. You will be able to access all logs.

**Reply by m1mir** [reply]
Thank you for the quick reply and the tips. I tried doing it with the options that you described, but the result hasn't changed.

The goal of my question was to figure out **how** heavy are the tests to run, so I could determine if I have access to a sufficiently strong computer "locally".
To me it seems that the reply is advice on how to run the tests given a limited hardware.
This is very useful advice, but it is an answer to a slightly different question.

**Reply by GaetanLepage** [reply]
Well, I don't know about the "minimal" hardware requirements.
I have a Ryzen 7 7840U with 32GB of RAM and I am able to run the tests fine.

CPU should not be a limit whatsoever, it should only impact the time the tests take to run.
As for memory, I would be surprised that the tests weren't working on a 16GB system... Nix should refrain itself to use all the available RAM.

How much memory do you have @traxys @MattSturgeon ?

**Reply by traxys** [reply]
Ryzen 7 5800X & 32GB

**Reply by MattSturgeon** [reply]
My primary machine is a laptop with 16GB. That struggles to run the tests most of the time, though there may be other issues on my end.

I also have a desktop with 64GB, but I don't use it much because I have yet to convert it over to nixos and I often run into weird quirks of using nix on another distro.

**Reply by MattSturgeon** [reply]
After running `tests` for a few minutes:

<details><summary>Details</summary>
<p>

![image](https://github.com/nix-community/nixvim/assets/5046562/99ab8d23-7e5f-409d-880a-a986adf98f49)

![image](https://github.com/nix-community/nixvim/assets/5046562/18825715-b0d6-41eb-8969-130edfb28232)


</p>
</details> 

EDIT: 30 mins later, it's still maxed out swap & ram, but it has finished evaluating and is now building!

**Reply by m1mir** [reply, marked as answer]
Thank you for the the information! I've managed to run the test suite on a laptop with similar properties to the one that Matt described and with NixOS with similar resource usage.

Based on what you all said and, based on the results it seems to me that **32GB memory**( meaning RAM *and* swap) is a practical minimum for running the checks locally.
